code/datums/components/gunpoint.dm

```diff
@@ -113,6 +117,8 @@
 
 ///Bang bang, we're firing a charged shot off
 /datum/component/gunpoint/proc/trigger_reaction()
+	SIGNAL_HANDLER_DOES_SLEEP
```

Comment:
This shouldn't be used on anything new. See if you can move the async parts to their own proc and call that asynchronously.

Reply:
Just for clarity, the PR is not adding this signal.

Reply:
Ah, nevermind then. I must have just missed it when I did my PR.